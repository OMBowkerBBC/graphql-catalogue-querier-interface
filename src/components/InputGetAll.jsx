import { useState } from 'react'
import Dropdown from '@bbc/igm-dropdown'
import '@bbc/igm-dropdown/dist/Dropdown.css'
import PropTypes from 'prop-types'

const options = [
  {
    value: 'episode',
    label: 'Episode'
  },
  {
    value: 'series',
    label: 'Series'
  },
  {
    value: 'brand',
    label: 'Brand'
  },
]

const InputGetAll = ({ setQueryArguments }) => {
  const [dropDownValue, setDropDownValue] = useState('episode')

  return (
    <Dropdown 
      options={options}
      placeholder='Choose an option...'
      size='small'
      onChange={(e) => {
        setDropDownValue(e)
        setQueryArguments({type: e.value})
      }}
      value={dropDownValue}
    />
  )
}

InputGetAll.propTypes = {
  setQueryArguments: PropTypes.func.isRequired,
}

export default InputGetAll
