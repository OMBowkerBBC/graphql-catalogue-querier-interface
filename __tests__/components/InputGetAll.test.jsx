import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import InputGetAll from '../../src/components/InputGetAll'

describe('InputGetAll test suite', () => {
  let mockedSetQueryArguments = vi.fn()

  describe('snapshot tests', () => {
    it('should match starting snapshot', () => {
      const tree = renderer.create(<InputGetAll setQueryArguments={vi.fn()} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  beforeEach(() => {
    render(<InputGetAll setQueryArguments={mockedSetQueryArguments} />)
  })

  it('should show default placeholder text', () => {
    expect(screen.getByText('Choose an option...')).toBeDefined()
  })

  it('should call setQueryArguments function when an option is pressed', async () => {
    fireEvent.click(screen.getByText('Choose an option...'))
    fireEvent.click(screen.getByText('Brand'))
    expect(mockedSetQueryArguments).toHaveBeenCalled()
  })

  it.each(['Episode', 'Series', 'Brand'])('should set only value to type %s when selected', (type) => {
    fireEvent.click(screen.getByText('Choose an option...'))
    fireEvent.click(screen.getByText(type))
    expect(screen.getByText(type)).toBeDefined()
    expect(() => screen.getByText('Choose an option...')).toThrow()
  })
})