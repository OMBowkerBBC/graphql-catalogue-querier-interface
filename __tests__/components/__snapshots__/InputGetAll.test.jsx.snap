// Vitest Snapshot v1

exports[`InputGetAll test suite > snapshot tests > should match starting snapshot 1`] = `
<div
  className="css-kvnfih"
>
  <button
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-label="Choose an option... - Select an option"
    className="css-lmeovg"
    id="dropdown-button-6"
    onClick={[Function]}
    onKeyDown={[Function]}
    size="small"
  >
    Choose an option...
    <div
      className="css-e0jyqw"
    >
      <svg
        focusable="false"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 29L32 3h-7.2L16 18.3 7.2 3H0l16 26z"
        />
      </svg>
    </div>
  </button>
  <div
    aria-live="polite"
  />
</div>
`;
